app/src/main/java/de/ur/explure/views/ProfileFragment.kt

```diff
@@ -1,6 +1,155 @@
+            viewModel.updateUserName(name.toString())
+        }
+        builder.setNegativeButton(String.format(getResources().getString(R.string.abort)), null)
+        builder.setCancelable(false)
+
+        val dialog = builder.create()
+        dialog.show()
+
+        dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = false
+
+        textInputEditText.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+                // do nothing
+            }
+
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                // do nothing
+            }
+
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                if (p0.isNullOrBlank()) {
+                    textInputLayout.error = String.format(getResources().getString(R.string.user_name_needed))
+                    dialog.getButton(AlertDialog.BUTTON_POSITIVE)
+                            .isEnabled = false
+                } else {
+                    textInputLayout.error = ""
+                    dialog.getButton(AlertDialog.BUTTON_POSITIVE)
+                            .isEnabled = true
+                }
+            }
+        })
+    }
+
+    private fun getEditUserNameLayout(context: Context): ConstraintLayout {
+        val constraintLayout = ConstraintLayout(context)
+        val layoutParams = ConstraintLayout.LayoutParams(
+                ConstraintLayout.LayoutParams.MATCH_PARENT,
+                ConstraintLayout.LayoutParams.WRAP_CONTENT
+        )
+        constraintLayout.layoutParams = layoutParams
+        constraintLayout.id = View.generateViewId()
+
+        val textInputLayout = TextInputLayout(context)
+        textInputLayout.boxBackgroundMode = TextInputLayout.BOX_BACKGROUND_OUTLINE
+        layoutParams.setMargins(
+                VERTICAL_MARGIN_EDIT_TEXT.toDp(context),
+                HORIZONTAL_MARGIN_EDIT_TEXT.toDp(context),
+                VERTICAL_MARGIN_EDIT_TEXT.toDp(context),
+                HORIZONTAL_MARGIN_EDIT_TEXT.toDp(context)
+        )
+        textInputLayout.layoutParams = layoutParams
+        textInputLayout.hint = String.format(getResources().getString(R.string.new_user_name_hint))
+        textInputLayout.id = View.generateViewId()
+        textInputLayout.tag = "textInputLayoutTag"
+
+        val textInputEditText = TextInputEditText(context)
+        textInputEditText.id = View.generateViewId()
+        textInputEditText.tag = "textInputEditTextTag"
+
+        textInputLayout.addView(textInputEditText)
+
+        val constraintSet = ConstraintSet()
+        constraintSet.clone(constraintLayout)
+
+        constraintLayout.addView(textInputLayout)
+        return constraintLayout
+    }
+
+    private fun Int.toDp(context: Context): Int = TypedValue.applyDimension(
+            TypedValue.COMPLEX_UNIT_DIP, this.toFloat(), context.resources.displayMetrics
+    ).toInt()
```

Comment:
Solche Extension-Functions gehören eigentlich nicht in Fragments oder Activities, sondern in separate Files ausgelagert mit der Endung 'Ext', damit klar ist, dass es sich um von dir hinzugefügte Erweiterungen handelt.